src/Lean/Meta/WHNF.lean

```diff
@@ -581,14 +581,43 @@ private def whnfDelayedAssigned? (f' : Expr) (e : Expr) : MetaM (Option Expr) :=
   else
     return none
 
-partial def expandLet (e : Expr) (vs : Array Expr) : Expr :=
-  if let .letE _ _ v b _  := e then
-    expandLet b (vs.push <| v.instantiateRev vs)
-  else if let some (#[], _, _, v, b) := e.letFunAppArgs? then
-    expandLet b (vs.push <| v.instantiateRev vs)
+/--
+Assuming `zeta` is enabled, zeta reduce lets. Does not take `zetaUnused` into account.
```

Comment:
This docstring seems outdated. `expandLet` is a pure function that does not take `zeta` nor `zetaUnused` as parameters.